virtual/environment_test.go

```diff
@@ -663,8 +663,8 @@ func TestHeartbeatAndSelfHealing(t *testing.T) {
 
 	// TODO: Sleeps in tests are bad, but I'm lazy to inject a clock right now and deal
 	//       with all of that.
-	require.NoError(t, env1.Close())
-	require.NoError(t, env2.Close())
+	require.NoError(t, env1.Close(context.Background()))
```

Comment:
I think the clean shutdown behavior is worth at least one test. I think the best way to exercise it would be to create a test that creates a variety of actors and whose Shutdown method implementation has some kind of side-effect. For example, for the Go actor you could make this method write a key to the KV store that indicates shutdown was called or increment some counter: https://github.com/richardartoul/nola/blob/master/virtual/environment_test.go#L930

then you can reopen the environment again with the same registry and if the clean shutdown was performed properly you should be able to make the actor read the key/counter that was written

Reply:
For the WASM actor you'll need to modify this file: https://github.com/richardartoul/nola/blob/master/testdata/tinygo/util/main.go

Note that since the WASM module is committed to the repo, after you modify it you'll need to run `make compile-wasm` before you'll be able to observe the change in the WASM module